src/sentry/event_manager.py

```diff
@@ -825,42 +823,39 @@ def _save_aggregate(self, event, hashes, release, **kwargs):
 
             group_is_new = False
 
+        # Keep a set of all of the hashes that are relevant for this event and
+        # belong to the destination group so that we can record this as the
+        # last processed event for each. (We can't just update every
+        # ``GroupHash`` instance, since we only want to record this for events
+        # that not only include the hash but were also placed into the
+        # associated group.)
+        relevant_group_hashes = set([instance for instance in all_hashes if instance.group_id == group.id])
+
         # If all hashes are brand new we treat this event as new
         is_new = False
-        new_hashes = [h[1] for h in all_hashes if h[0] is None]
+        new_hashes = [h for h in all_hashes if h.group_id is None]
         if new_hashes:
-            # This is an attempt at optimizing the lookup against GroupHash
-            # Previously this was doing something like:
-            # UPDATE "sentry_grouphash" SET "group_id" = x WHERE ("sentry_grouphash"."project_id" = x AND "sentry_grouphash"."hash" IN ('x') AND "sentry_grouphash"."group_id" IS NULL)
-            # This query is causing the query planner to choose the sentry_grouphash_group_id index
-            # rather than the expected sentry_grouphash_project_id_4a293f96a363c9a2_uniq index.
-            # So this change to a SELECT/UPDATE instead forces us to hit the expected index.
-            # We then filter in python to check if we need to update, and we update back explicitly
-            # on the primary keys.
-            affected = map(
-                # Extract just the id
-                lambda x: x[0],
-                filter(
-                    # find the rows that have no group id
-                    lambda x: x[1] is None, list(
-                        GroupHash.objects.filter(
-                            project=project,
-                            hash__in=new_hashes,
-                        ).values_list('id', 'group_id')
-                    )
-                )
-            )
-
-            if affected:
-                GroupHash.objects.filter(
-                    id__in=affected,
-                ).update(group=group)
-
-            if len(affected) != len(new_hashes):
-                self._ensure_hashes_merged(group, new_hashes)
```

Comment:
This was already subject to a race condition that would make it unreliable, so I removed it. I don't feel good about this.